===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ _FtpTelnetReload:n$53 == { -1 } }) free(*(_FtpTelnetReload:pPolicyConfig)) at 29 (line 549, column 9) }
===== Actual patch in diff format =====
--- a/table1/benchmarks/snort-2.9.13/src/dynamic-preprocessors/ftptelnet/spp_ftptelnet.c
+++ b/table1/benchmarks/snort-2.9.13/src/dynamic-preprocessors/ftptelnet/spp_ftptelnet.c
@@ -541,7 +541,7 @@ static void _FtpTelnetReload(struct _SnortConfig *sc, tSfPolicyUserContextId ftp
                                             "FTP/Telnet configuration.\n");
         }
 
-        sfPolicyUserDataSetCurrent(ftp_telnet_swap_config, pPolicyConfig);
+        int tmp = sfPolicyUserDataSetCurrent(ftp_telnet_swap_config, pPolicyConfig);
 
         iRet = FtpTelnetInitGlobalConfig(pPolicyConfig,
                                          ErrorString, iErrStrLen);
@@ -559,6 +559,8 @@ static void _FtpTelnetReload(struct _SnortConfig *sc, tSfPolicyUserContextId ftp
                         NULL, NULL, NULL, NULL, NULL);
             }
         }
+        if (tmp == -1)
+            free(pPolicyConfig);
     }
     else if (strcasecmp(pcToken, TELNET) == 0)
     {
